packages/react/scripts/generatePrimitivesCatalog.ts

```diff
@@ -115,10 +115,20 @@ const getCatalogComponentProperty = (
   } else if (propType.isNumber() || propType.isNumberLiteral()) {
     return { type: PrimitiveCatalogComponentPropertyType.Number };
   } else if (propType.isUnion()) {
+    const hasNumber = propType
+      .getUnionTypes()
+      .every((prop) => prop.isNumber() || prop.isNumberLiteral());
```

Comment:
This should be `.every` and not `.some` like in the `hasString` check below?

Reply:
With 'some' it was falsely changing lots of types to "number" when the union included number. We only want to catch union types of all number types here.